include/pinocchio/multibody/joint/joint-mimic.hpp

```diff
@@ -688,31 +780,34 @@ namespace pinocchio
       const bool update_I) const
     {
       // TODO: fixme
+      assert(
+        false
+        && "Joint Mimic is not supported for aba yet. Remove it from your model if you want to use "
+           "this function");
       m_jmodel_ref.calc_aba(
         data.m_jdata_ref, armature, PINOCCHIO_EIGEN_CONST_CAST(Matrix6Like, I), update_I);
     }
 
     static std::string classname()
     {
-      return std::string("JointModelMimic<") + JointModel::classname() + std::string(">");
-      ;
+      return std::string("JointModelMimic");
     }
 
     std::string shortname() const
     {
-      return std::string("JointModelMimic<") + m_jmodel_ref.shortname() + std::string(">");
+      return classname();
     }
 
     /// \returns An expression of *this with the Scalar type casted to NewScalar.
     template<typename NewScalar>
-    typename CastType<NewScalar, JointModelMimic>::type cast() const
+    typename CastType<NewScalar, JointModelMimicTpl>::type cast() const
     {
-      typedef typename CastType<NewScalar, JointModelMimic>::type ReturnType;
+      typedef typename CastType<NewScalar, JointModelMimicTpl>::type ReturnType;
 
       ReturnType res(
-        m_jmodel_ref.template cast<NewScalar>(), pinocchio::cast<NewScalar>(m_scaling),
-        pinocchio::cast<NewScalar>(m_offset));
-      res.setIndexes(id(), idx_q(), idx_v());
+        m_jmodel_ref.template cast<NewScalar>(), ScalarCast<NewScalar, Scalar>::cast(m_scaling),
+        ScalarCast<NewScalar, Scalar>::cast(m_offset));
+      res.setIndexes(id(), idx_q(), idx_v(), idx_j());
```

Comment:
Isn't a `setMimicIndices` missing ?
See https://github.com/abussy-aldebaran/pinocchio/commit/5a8f2050b4785fa7ced21f513146461e09a15901
and https://github.com/abussy-aldebaran/pinocchio/commit/afef317dbdb5721ba59042b3e9e78b2e0218c71b